internal/testutils/xds/e2e/clientresources.go

```diff
@@ -523,28 +524,71 @@ func ClusterResourceWithOptions(opts *ClusterOptions) *v3clusterpb.Cluster {
 	return cluster
 }
 
+// EndpointOptions contains options to configure an Endpoint (or
+// ClusterLoadAssignment) resource.
+type EndpointOptions struct {
+	// ClusterName is the name of the Cluster resource (or EDS service name)
+	// containing the endpoints specified below.
+	ClusterName string
+	// Host is the hostname of the endpoints. In our e2e tests, hostname must
+	// always be "localhost".
+	Host string
+	// Ports is a set of ports on "localhost" where the endpoints corresponding
+	// to this resource reside.
+	Ports []uint32
+	// DropPercents is a map from drop category to a drop percentage.
```

Comment:
Nit: for your other resource creation prs, you briefly mentioned what happened if these fields aren't set (default) in this case map is nil or empty. If this isn't set, cla.Policy is written to, if not, it is not written to. Please briefly describe downstream behavior of what occurs if this is set or not.

Reply:
Done.